### DISCUSSION TITLE
Adding unit tests

### DISCUSSION THREAD

**Discussion by NicolasGensollen** [aramis-lab/clinica#595] [General]
I believe this would be a great addition and allow us to work in a more TDD fashion.
If we can decide on where these tests should be implemented and how they should be organized, I'll get started asap with tests for #587 

I imagine two main different ways we could organize them:

- either we group them all in `test/unittests/{engine,iotools,utils...}`
- or we can put them in a `tests`folder in each module (for example: `Clinica/iotools/converters/adni_to_bids/tests/`

Thoughts?

**Comment by ghisvail** [comment]
Since we have already adopted the convention of storing tests out-of-tree, I'd be in favour of keeping it for unit tests as well.

**Reply by omar-rifai** [reply]
Agreed. As discussed, we can mirror the pipelines hierarchy in a seperate `test/unitest/` directory.
